adsingestp/parsers/jats.py

```diff
@@ -912,23 +913,29 @@ def citation_context(
 
         self.back_meta = document.back
         body = document.body
-        xrefs = body.find_all("xref")
+        xrefs = body.find_all("xref", attrs={"ref-type": "bibr"})
+
         cites = {}  # {rid_1: ["context 1", "context 2", ...]}
         for x in xrefs:
             id = x["rid"]
             immediate_para = x.find_parent("p")  # try to find the containing paragraph
             if immediate_para:
-                context = immediate_para.get_text()
-                if len(context) < num_char:
-                    prev_para = immediate_para.find_previous_sibling("p")
-                    if prev_para:
-                        context = prev_para.get_text() + context
-                    next_para = immediate_para.find_next_sibling("p")
-                    if next_para:
-                        context = context + next_para.get_text()
+                context = immediate_para
+                if text_output:
+                    context = immediate_para.get_text()
+                    if len(context) < num_char:
+                        prev_para = immediate_para.find_previous_sibling("p")
+                        if prev_para:
+                            context = prev_para.get_text() + context
+                        next_para = immediate_para.find_next_sibling("p")
+                        if next_para:
+                            context = context + next_para.get_text()
```

Comment:
Ok, I've figured out how to fix the unittesting, but it required a rethink about the output of this function, if `text_output=False`. Right now, in that case, you're returning BeautifulSoup tag objects, but actually, I think you really want to return the raw XML as a string instead. Small, but important, difference. The output will look the same, but it'll behave differently, and if you end up needing to have it as a BeautifulSoup object, you can convert it.

To output the raw XML string, add this line (it'll be the pair for the `if text_output:` on line 924:
```
else:
    context = str(context)
```

And see my comments on test_jats.py for how to implement this there.